CrossPlatformDesktopProject/CrossPlatformDesktopProject/CollisionHandler/CollisionResponse.cs

```diff
@@ -0,0 +1,117 @@
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CrossPlatformDesktopProject.CollisionHandler
+{
+    class CollisionResponse
+    {
+        private Dictionary<Tuple<Type, Type, CollisionSides>, Type> commandMap;
+        private HashSet<Tuple<Type, Type, CollisionSides>> keySet;
+
+        public CollisionResponse()
+        {
+            commandMap = new Dictionary<Tuple<Type, Type, CollisionSides>, Type>();
+            BuildDictionary();
+            keySet = new HashSet<Tuple<Type, Type, CollisionSides>>(commandMap.Keys);
+        }
+
+        private void BuildDictionary()
+        {
+            Type playerType = typeof(Player);
+            
+            //Enemy Types
+            Type batType = typeof(Bat);
+            Type bossType = typeof(Boss);
+            Type gelType = typeof(Gel);
+            Type goriyaType = typeof(Goriya);
+            Type skeletonType = typeof(Skeleton);
+            Type[] enemyTypes = { batType, bossType, gelType, goriyaType, skeletonType };
+
+            //Object Types
+            Type boomerangType = typeof(Boomerang);
+            Type bombType = typeof(Bomb);
+            Type bowType = typeof(Gel);
+            Type[] objectTypes = { boomerangType, bowType, bombType };
+
+            //Obstacle Types
+            var typeOfObstacle = typeof(IObstacle);
+            var obstacleTypes = AppDomain.CurrentDomain.GetAssemblies()
+                .SelectMany(s => s.GetTypes())
+                .Where(p => typeOfObstacle.IsAssignableFrom(p));
+
+            //Enemy on player
+            foreach (Type enemySubject in enemyTypes)
+            {
+                commandMap.Add(new Tuple<Type,Type,CollisionSides>(enemySubject, playerType, CollisionSides.Down), typeof(TakeDamageCommand));
+                commandMap.Add(new Tuple<Type, Type, CollisionSides>(enemySubject, playerType, CollisionSides.Left), typeof(TakeDamageCommand));
+                commandMap.Add(new Tuple<Type, Type, CollisionSides>(enemySubject, playerType, CollisionSides.Right), typeof(TakeDamageCommand));
+                commandMap.Add(new Tuple<Type, Type, CollisionSides>(enemySubject, playerType, CollisionSides.Up), typeof(TakeDamageCommand));
+            }
+            foreach(Type obstacleSubject in obstacleTypes)
+            {
+                commandMap.Add(new Tuple<Type, Type, CollisionSides>(obstacleSubject, playerType, CollisionSides.Down), typeof(ResetCommand));
+                commandMap.Add(new Tuple<Type, Type, CollisionSides>(obstacleSubject, playerType, CollisionSides.Left), typeof(ResetCommand));
+                commandMap.Add(new Tuple<Type, Type, CollisionSides>(obstacleSubject, playerType, CollisionSides.Right), typeof(ResetCommand));
+                commandMap.Add(new Tuple<Type, Type, CollisionSides>(obstacleSubject, playerType, CollisionSides.Up), typeof(ResetCommand));
+            }
```

Comment:
I think both of these loops could be contained in a bigger loop to "Don't Repeat Yourself"
as in 

```C#
foreach(CollisionSides side in Enum.GetValues(typeof(CollisionSides))
{
    // foreach(enemySubject ...) ...
    // foreach(obstacleSubject ...) ...
}
```

Reply:
There may be cases where we have different behaviors based on direction, so I have avoided that for the time being. After we know what we need for every interaction it'll be easier to simplify.